federation/cmd/federation-controller-manager/app/controllermanager.go

```diff
@@ -54,6 +58,13 @@ import (
 	"k8s.io/client-go/discovery"
 )
 
+const (
+	apiserverWaitTimeout   = 30 * 2 * time.Second
+	apiserverRetryInterval = 1 * time.Second
```

Comment:
We should be initializing the controller-manager as early as possible. controller-manager depends on api-server and they both are started independently. So either one may start functioning before the other. If we introduce exponential backoff, i am worried it may induce unnecessary delay in initializing controller-manager. Also it is not the critical path and we are just in initialization stage.
I would increase the apiserverRetryInterval to 2 seconds. and the apiserverWaitTimeout to 2 minutes. Is this fine?